src/Vault.sol

```diff
@@ -432,6 +435,13 @@ contract Vault {
 
         uint256 assetAddressesLength = assetAddresses.length;
         for (uint256 i; i < assetAddressesLength;) {
+            if (assetAmounts[i] == 0) {
```

Comment:
Else it may push the address to the list of addresses stored, even though it has a zero balance (and will do so on other subsequent deposits, so you could have 2x the same erc20 address in the stored addresses, counting assets double)

Reply:
Maybe add as comment?
Was not trivial to me